qiskit_experiments/library/randomized_benchmarking/rb_utils.py

```diff
@@ -294,3 +310,264 @@ def calculate_2q_epg(
+        return self._hash
+
+    def __eq__(self, other):
+        return self._qubits == other._qubits and self._gate == other._gate
+
+    def __repr__(self):
+        qs = ", ".join(map(str, sorted(self._qubits)))
+        return f'(({qs}), "{self._gate}")'
+
+
+def lookup_epg_ratio(gate: str, n_qubits: int) -> Union[None, int]:
+    """Look-up preset gate error ratio for given basis gate name.
+
+    In the table the error ratio is defined based on the count of
+    typical assembly gate in the gate decomposition.
+    For example, "u3" gate can be decomposed into two "sx" gates.
+    In this case, the ratio of "u3" gate error becomes 2.
+
+    .. note::
+
+        This table is not aware of the actual waveform played on the hardware,
+        and the returned error ratio is just a guess.
+        To be precise, user can always set "gate_error_ratio" option of the experiment.
+
+    Args:
+        gate: Name of the gate.
+        n_qubits: Number of qubits measured in the RB experiments.
+
+    Returns:
+        Corresponding error ratio.
+
+    Raises:
+        QiskitError: When number of qubit is more than three.
+    """
+
+    # Gate count in (X, SX)-based decomposition. VZ gate contribution is ignored.
+    # Amplitude or duration modulated pulse implementation is not considered.
+    standard_1q_ratio = {
+        "u1": 0.0,
+        "u2": 1.0,
+        "u3": 2.0,
+        "u": 2.0,
+        "p": 0.0,
+        "x": 1.0,
+        "y": 1.0,
+        "z": 0.0,
+        "t": 0.0,
+        "tdg": 0.0,
+        "s": 0.0,
+        "sdg": 0.0,
+        "sx": 1.0,
+        "sxdg": 1.0,
+        "rx": 2.0,
+        "ry": 2.0,
+        "rz": 0.0,
+        "id": 0.0,
+        "h": 1.0,
+    }
+
+    # Gate count in (CX, CSX)-based decomposition, 1q gate contribution is ignored.
+    # Amplitude or duration modulated pulse implementation is not considered.
+    standard_2q_ratio = {
+        "swap": 3.0,
+        "rxx": 2.0,
+        "rzz": 2.0,
+        "cx": 1.0,
+        "cy": 1.0,
+        "cz": 1.0,
+        "ch": 1.0,
+        "crx": 2.0,
+        "cry": 2.0,
+        "crz": 2.0,
+        "csx": 1.0,
+        "cu1": 2.0,
+        "cp": 2.0,
+        "cu": 2.0,
+        "cu3": 2.0,
+    }
+
+    if n_qubits == 1:
+        return standard_1q_ratio.get(gate, None)
+
+    if n_qubits == 2:
+        return standard_2q_ratio.get(gate, None)
+
+    raise QiskitError(
+        f"Standard gate error ratio for {n_qubits} qubit RB is not provided. "
+        "Please explicitly set 'gate_error_ratio' option of the experiment."
+    )
+
+
+def calculate_epg(
+    epc: Union[float, uncertainties.UFloat],
+    qubits: Sequence[int],
+    gate_error_ratio: Dict[str, float],
+    gate_counts_per_clifford: Dict[QubitGateTuple, float],
+) -> Dict[str, Union[float, uncertainties.UFloat]]:
+    r"""Compute EPGs of basis gates from fit EPC value.
+
+    This function just distributes the measured EPC :math:`\cal E` into basis gates :math:`\{g_i\}`
```

Comment:
It would be good for this documentation to be located somewhere else where its more likely to be seen, perhaps the RB tutorial would be a good place?